src/nvs.rs

```diff
@@ -746,10 +746,7 @@ impl<T: NvsPartitionId> EspNvs<T> {
     /// This function will return an error if
     /// - there is no memory available for allocation of internal structures
     /// - for some reason the [`EspNvs::handle`] is invalid (should not happen)
-    #[cfg(all(
-        esp_idf_version_major = "5",
-        not(any(esp_idf_version_minor = "1", esp_idf_version_minor = "0"))
-    ))]
+    #[cfg(any(esp_idf_version_patch_at_least_5_2_0, esp_idf_version_at_least_5_2_0))]
```

Comment:
You don't need `esp_idf_version_patch_at_least_5_2_0`. for your particular case - where the functionality is available for ALL ESP IDFs >= 2.2.0 - `esp_idf_version_at_least_5_2_0` is sufficient.

The `esp_idf_version_patch_at_least_*` variants are useful only if Espressif decides to back-port some functionality in an earlier ESP-IDF release. Say, in ESP IDF 5.1 and it will pop-up there after patch 5. Then you need to do `any(esp_idf_version_patch_at_least_5_1_5, esp_idf_version_at_least_5_2_0)` 
which means "either from V5.1.5 till V5.1.max, OR from V5.2.2 onwards".